Makefile

```diff
@@ -371,10 +371,11 @@ endif
 
 LIBOBJECTS += $(TOOL_LIB_SOURCES:.cc=.o)
 MOCKOBJECTS = $(MOCK_LIB_SOURCES:.cc=.o)
+TOOLLIBOBJECTS = $(TOOL_LIB_SOURCES:.cc=.o)
 
 GTEST = $(GTEST_DIR)/gtest/gtest-all.o
 TESTUTIL = ./util/testutil.o
-TESTHARNESS = ./util/testharness.o $(TESTUTIL) $(MOCKOBJECTS) $(GTEST)
+TESTHARNESS = ./util/testharness.o $(TESTUTIL) $(MOCKOBJECTS) $(TOOLLIBOBJECTS) $(GTEST)
```

Comment:
I looked at the Makefile more closely just now. TESTHARNESS seems to be linked to all the tests (100s of them). And now we would link the tool objects to all the tests with this change. 
Would it be better to explicitly link the ldb_tool.o (or whatever is needed) to unity_test? It would be much more contained change.